### PR TITLE
Evaluate absolute directory when user run playground

### LINKED ISSUE #619: Tiup playground can't use `~` as directory
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?

tiup playground --db.config=~/cp.toml

2. What did you expect to see?
Start success.

3. What did you see instead?
```
➜  tidb git:(master) tiup playground --db.config=~/cp.toml
Starting component `playground`:  --db.config=~/cp.toml
Use latest stable version: v4.0.2

	You can specified the version by running "tiup playground <version>", example: "tiup playground v4.0.0"

Playground Bootstrapping...
Start pd instance...
Start tikv instance...
Start tidb instance...
............................................................To view the dashboard: http://127.0.0.1:2379/dashboard
To view the Prometheus: http://127.0.0.1:9090
To view the Grafana: http://127.0.0.1:3000
^CGot signal interrupt (Component: playground ; PID: 80635)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1ae2512]

goroutine 22 [running]:
github.com/pingcap/tiup/components/playground/instance.(*TiFlashInstance).Pid(0xc0002bc820, 0x1e03e1c)
	<autogenerated>:1 +0x32
main.(*Playground).terminate.func1(0x1e03e1c, 0x7, 0x1fccfe0, 0xc0002bc820, 0x0, 0x0)
	github.com/pingcap/tiup@/components/playground/playground.go:898 +0x1f4
main.(*Playground).RWalkInstances(0xc0002ba000, 0xc000383f90, 0x8000, 0xc00025b790)
	github.com/pingcap/tiup@/components/playground/playground.go:493 +0x124
main.(*Playground).terminate(0xc0002ba000, 0x2, 0xc00037c000, 0x2, 0x2)
	github.com/pingcap/tiup@/components/playground/playground.go:891 +0x68
created by main.(*Playground).bootCluster.func3
	github.com/pingcap/tiup@/components/playground/playground.go:844 +0x1df
Component `playground` exit with error: exit status 2
```

4. What version of TiUP are you using (`tiup --version`)?
➜  tidb git:(master) tiup --version
v1.0.8 tiup
Go Version: go1.13
Git Branch: master
GitHash: 4276089

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <heng@lonng.org>

### What problem does this PR solve? 

Fix #619

### CODE REVIEW

File: components/playground/playground.go [hunk 1]
```diff
@@ -359,7 +360,36 @@ func (p *Playground) handleScaleIn(w io.Writer, pid int) error {
 	return nil
 }
 
-func (p *Playground) sanitizeConfig(boot instance.Config, cfg *instance.Config) {
+func (p *Playground) absDir(dir string) (string, error) {
+	if dir == "" {
+		return "", nil
+	}
+
+	if !filepath.IsAbs(dir) && !strings.HasPrefix(dir, "~/") {
```
Comment (AstroProfundis) [inline components/playground/playground.go:368]: The `~` should be expanded by the shell in most cases. It's ok to keep this logic, I'm just noting that for records.

File: components/playground/playground.go [hunk 2]
```diff
@@ -359,7 +360,36 @@ func (p *Playground) handleScaleIn(w io.Writer, pid int) error {
 	return nil
 }
 
-func (p *Playground) sanitizeConfig(boot instance.Config, cfg *instance.Config) {
+func (p *Playground) absDir(dir string) (string, error) {
```
Comment (july2993) [inline components/playground/playground.go:363]: could you add some test for this?

File: components/playground/playground.go [hunk 3]
```diff
@@ -359,7 +360,36 @@ func (p *Playground) handleScaleIn(w io.Writer, pid int) error {
...
+	if dir == "" {
+		return "", nil
+	}
+
+	if !filepath.IsAbs(dir) && !strings.HasPrefix(dir, "~/") {
+		wd := os.Getenv(localdata.EnvNameWorkDir)
```
Comment (july2993) [inline components/playground/playground.go:369]: this will be the dir i run `tiup ...` ?

File: components/playground/playground.go (hunk 2 above)
Comment (lucklove) [inline components/playground/playground.go:363]: Can you add a test for this function? I'm not sure if dir like `$HOME/a/b/c` works

File: components/playground/playground.go (hunk 1 above)
Comment (lonng) [reply components/playground/playground.go:368]: `tiup playground --db.config ~/db.toml` will expand but `tiup playground --db.config=~/db.toml` will not.

File: components/playground/playground.go (hunk 3 above)
Comment (lonng) [reply components/playground/playground.go:369]: Yes

File: components/playground/playground.go (hunk 2 above)
Comment (lonng) [reply components/playground/playground.go:363]: OK

Comment (lonng) [reply components/playground/playground.go]: Fixed in c0de8f1ffeb5affa75bc680814a7a6f8f68a6901

**Pull request merged**